sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/CreateTables.java

```diff
@@ -78,20 +80,38 @@ CreateTables withTestServices(BigQueryServices bqServices) {
   @Override
   public PCollection<KV<TableDestination, TableRow>> expand(
       PCollection<KV<TableDestination, TableRow>> input) {
+    List<PCollectionView<Map<String, String>>> sideInputs = Lists.newArrayList();
+    if (schemaFunction.getSideInput() != null) {
+      sideInputs.add(schemaFunction.getSideInput());
+    }
     return input.apply(
         ParDo.of(
             new DoFn<KV<TableDestination, TableRow>, KV<TableDestination, TableRow>>() {
+              SchemaFunction schemaFunctionCopy;
+
+              @StartBundle
+              public void startBundle(Context c) {
+                this.schemaFunctionCopy = SerializableUtils.clone(schemaFunction);
+              }
+
               @ProcessElement
               public void processElement(ProcessContext context)
                   throws InterruptedException, IOException {
+                if (schemaFunctionCopy.getSideInput() != null) {
```

Comment:
I don't think this is correct. It's allowed to process elements in several different windows, or from several different keys, within a single bundle - the value of the side input will be different. If that is disallowed, please clearly document that, and if possible, assert it at runtime (or is it possible to structure the transform so that it's impossible? e.g. rewindow the side input into the global window, or require that it's already globally windowed?..)